ui/server/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import java.io.File
+
+import akka.http.scaladsl.server.directives.{Credentials, SecurityDirectives}
+import akka.http.scaladsl.server.directives.Credentials.Provided
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.ui.security.api.{LoggedUser, Permission}
+import pl.touk.nussknacker.ui.security.basicauth.BasicHttpAuthenticator
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class OAuth2Authenticator extends SecurityDirectives.AsyncAuthenticator[LoggedUser] with LazyLogging {
+  def apply(credentials: Credentials): Future[Option[LoggedUser]] = Future {
+    authorize(credentials)
+  }
+
+  private[security] def authorize(credentials: Credentials): Option[LoggedUser] = {
+    logger.debug(s"Trying authorize.. $credentials")
+    credentials match {
+      case c @ Provided(token) => Option.apply(LoggedUser(token, Map("Default" -> Permission.values)))
```

Comment:
token is not an id (LoggedUser)? this token is an accessToken but we may need refreshToken later on?

Reply:
This is only for tests.. :) In future this case will take token and transfer him to service who will be try get user information from thirdpart api.